[[ Bug 19679 ]] Use bfd linker on android

livecodeali opened the pull request:
When the gold linker is used to link Standalone-Community
(librevandroid.so) the dynamic section is ordered differently
which appears to cause a 'missing PT_DYNAMIC' error when building
android standalones. Using the bfd linker instead fixes this issue.

livecodeali committed:
[[ Bug 19679 ]] Use bfd linker on android
When the gold linker is used to link Standalone-Community
(librevandroid.so) the dynamic section is ordered differently
which appears to cause a 'missing PT_DYNAMIC' error when building
android standalones. Using the bfd linker instead fixes this issue.

runrevmark commented:
@livecodeali This rings a bell insofar as I recall seeing an option to set the linker somewhere else. Must have got removed or something by an unsuspecting PR :)

@livecode-vulcan review ok f63819b

livecode-vulcan commented:
💙 review by @runrevmark ok f63819bdf7c2c8b05ec622be069f992be8b2fdf9

livecode-vulcan commented:
😞 test failure f63819bdf7c2c8b05ec622be069f992be8b2fdf9

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/230)
* try-community-armv6-android-api9: [failure](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/238)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/218)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/162)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/162)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/228)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/233)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/232)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/236)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/232)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/236)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/256)
* try-community-x86_64-win32: [pending](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/227)

runrevmark committed:
[[ BUg 19679 ]] Add missing '-'

This patch corrects the 'use-ld=bfd' option in config/android-settings.gypi,
making sure it begins with '-f' and not just 'f'.

runrevmark commented:
@livecode-vulcan self review ok ff68218

runrevmark commented:
@livecode-vulcan review ok ff6e218

livecode-vulcan commented:
💙 review by @runrevmark ok ff6e218da850883155883431c6a35b6271b1395d

livecode-vulcan commented:
😎 test success ff6e218da850883155883431c6a35b6271b1395d

* try-community-armv6-android-api8: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api8/builds/233)
* try-community-armv6-android-api9: [success](https://vulcan.livecode.com/builders/try-community-armv6-android-api9/builds/241)
* try-community-js-emscripten-sdk1.35: [success](https://vulcan.livecode.com/builders/try-community-js-emscripten-sdk1.35/builds/221)
* try-community-universal-ios-iphoneos10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphoneos10.3/builds/165)
* try-community-universal-ios-iphonesimulator10.3: [success](https://vulcan.livecode.com/builders/try-community-universal-ios-iphonesimulator10.3/builds/165)
* try-community-universal-mac-macosx10.6: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.6/builds/231)
* try-community-universal-mac-macosx10.9: [success](https://vulcan.livecode.com/builders/try-community-universal-mac-macosx10.9/builds/236)
* try-community-x86-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian7/builds/235)
* try-community-x86-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86-linux-debian8/builds/239)
* try-community-x86_64-linux-debian7: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian7/builds/235)
* try-community-x86_64-linux-debian8: [success](https://vulcan.livecode.com/builders/try-community-x86_64-linux-debian8/builds/239)
* try-community-x86-win32: [success](https://vulcan.livecode.com/builders/try-community-x86-win32/builds/259)
* try-community-x86_64-win32: [success](https://vulcan.livecode.com/builders/try-community-x86_64-win32/builds/230)
